src/components/Agent/AgentTable.tsx

```diff
@@ -79,19 +84,64 @@ const AgentTable: React.FC<AgentTableProps> = ({ obj }) => {
 
     return restBmh;
   });
-  const mergedHosts = [...restHosts, ...restBmhs];
+  */
+  const mergedHosts = [...restHosts]; //, ...restBmhs];
 
   return (
-    <Stack className="agent-table">
-      <StackItem>
-        <HostsTable
-          hosts={mergedHosts}
-          EmptyState={() => <div>empty</div>}
-          columns={getColumns()}
-        />
-      </StackItem>
-    </Stack>
+    <>
+      <Stack className="agent-table">
+        <StackItem>
+          {loaded ? (
+            <HostsTable
+              hosts={mergedHosts}
+              EmptyState={() => <div>no hosts</div>}
+              columns={getColumns()}
+              canEditHost={() => true}
+              onEditHost={(host, inventory) =>
+                editHostModal.open({
+                  host,
+                  inventory,
+                  usedHostnames: [],
+                  onSave: async ({ hostname, hostId }) => {
+                    const host = hosts.find((h) => h.metadata.uid === hostId);
+                    await k8sPatch(agentModel, host, [{
+                      op: 'add',
+                      path: `/spec/hostname`,
+                      value: hostname,
+                    },
+                    {
+                      op: 'replace',
+                      path: `/spec/approved`,
+                      value: true,
+                    }]);
+                  },
+                })
+              }
+              canEditRole={() => true}
+              onEditRole={async (host, role) => {
+                const agent = hosts.find((h) => h.metadata.uid === host.id);
+                await k8sPatch(agentModel, agent, [{
+                  op: 'replace',
+                  path: `/spec/role`,
+                  value: role,
+                }]);
+              }}
+            />
+          ) : (
+            <Bullseye>
+              <Spinner isSVG />
+            </Bullseye>
```

Comment:
There is a `LoadingState` component in the ui-lib which you could use here

Reply:
updated to `LoadingState`